yt_dlp/YoutubeDL.py

```diff
@@ -2641,6 +2641,16 @@ def _fill_common_fields(self, info_dict, final=True):
             if final and info_dict.get('%s_number' % field) is not None and not info_dict.get(field):
                 info_dict[field] = '%s %d' % (field.capitalize(), info_dict['%s_number' % field])
 
+        deprecated_multivalue_fields = {
+            'artist': 'artists',
+            'composer': 'composers',
+            'album_artist': 'album_artists',
+            'genre': 'genres',
+        }
+        for deprecated_field, new_field in deprecated_multivalue_fields.items():
+            if info_dict.get(deprecated_field):
+                info_dict[new_field] = re.split(r', ?', info_dict[deprecated_field])
```

Comment:
I found a new issue. These use-cases will break. We need a way to handle it

https://github.com/yt-dlp/yt-dlp/blob/85b33f5c163f60dbd089a6b9bc2ba1366d3ddf93/README.md#L1781-L1785

I'm not sure what's the best soln, but this should work:
```suggestion
        for deprecated_field, new_field in deprecated_multivalue_fields.items():
            if info_dict.get(deprecated_field):
                info_dict[new_field] = re.split(r', ?', info_dict[deprecated_field])
            elif info_dict.get(new_field):
                info_dict[depecated_field] = ', '.join(new_field)
```

and then revert

```diff
-        add('artist', ('artists', 'creator', 'uploader', 'uploader_id'))
+        add('artist', ('artist', 'creator', 'uploader', 'uploader_id'))
```

Opinion?

Reply:
I think we should use the new fields whenever possible. So I don't agree with this change:
> ```diff
> -        add('artist', ('artists', 'creator', 'uploader', 'uploader_id'))
> +        add('artist', ('artist', 'creator', 'uploader', 'uploader_id'))
> ```
We marked `artist` as being deprecated. It's strange to use deprecated fields in the core code.


> I found a new issue. These use-cases will break. We need a way to handle it.

I rewrote your suggestion with walrus operators. It looks pretty neat.
```suggestion
        for deprecated_field, new_field in deprecated_multivalue_fields.items():
            if deprecated_value := info_dict.get(deprecated_field):
                info_dict[new_field] = re.split(r', ?', deprecated_value)
            elif new_value := info_dict.get(new_field):
                info_dict[deprecated_field] = new_value.join(', ')
```
I'll update README to reflect changes in field names. `%(artists)s` and `%(artist)s` yield the same result, so it wouldn't be a problem.

Reply:
> I think we should use the new fields whenever possible. So I don't agree with this change:
> 
> > ```diff
> > -        add('artist', ('artists', 'creator', 'uploader', 'uploader_id'))
> > +        add('artist', ('artist', 'creator', 'uploader', 'uploader_id'))
> > ```

I don't like this either. But idk what would be a better solution.

In case the issue wasn't clear: Imagine `artists = ['a', 'b']` and user sets `artist=c` by `--parse-metadata`. Then the metadata embedded must be `c`, not `a, b`

Reply:
Ok, got it. You're right, we'll need to stick with deprecated fields for a while. I'll implement the changes in `FFMpegMetadataPP` according to your suggestions.